File: cs/BootCampCode/testcases/chapterSix/ShortCutAdjustmentTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using OpenQA.Selenium.Support.UI;

namespace BootCampCode.testcases.chapterSix
{
    [TestClass]
    public class ShortCutAdjustmentTest
    {
        [TestMethod]
        public void ShortCutAdjustment()
        {
            IWebDriver driver = new ChromeDriver();

            WebDriverWait myWaitVar = new WebDriverWait(driver, TimeSpan.FromSeconds(20));

            // Open the website
            driver.Navigate().GoToUrl("https://techblog.polteq.com/testshop/index.php");

            driver.Manage().Window.Maximize();

            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(10));
            IWebElement signInButton = driver.FindElement(By.ClassName("login"));
            if (signInButton.Displayed)
            {
                signInButton.Click();
                IWebElement email = driver.FindElement(By.Id("email"));
                IWebElement password = driver.FindElement(By.Id("passwd"));
                email.Clear();
                email.SendKeys("bootcamper@feelthepain.com");
                password.Clear();
                password.SendKeys("1qazxsw2");

                IWebElement submitSignIn = driver.FindElement(By.Id("SubmitLogin"));
                submitSignIn.Click();
            }

            //Go to the homepage
            driver.FindElement(By.CssSelector("img.logo")).Click();

            //personal info page can be found below
            driver.FindElement(By.CssSelector("a[title='Manage my personal information']")).Click();

            IWebElement fNameField = driver.FindElement(By.Id("firstname"));
            String fName = fNameField.GetAttribute("value");
            Console.WriteLine(fName);
            String newName = "General";
            if (fName == "Seargeant")
            {
                fNameField.Click();
                fNameField.Clear();
                fNameField.SendKeys(newName);

            }
            else if (fName == newName || fName != "Seargeant")
            {
                fNameField.Click();
                fNameField.Clear();
                fNameField.SendKeys("Seargeant");
            }

            IWebElement currentPwField = driver.FindElement(By.Id("old_passwd"));
            String currentPw = "1qazxsw2";
            currentPwField.Click();
            currentPwField.SendKeys(currentPw);

            IWebElement saveButton = driver.FindElement(By.XPath("//button//span[contains(text(), 'Save')]"));
            saveButton.Click();

            IWebElement saveIsSuccesful = wait.Until(ExpectedConditions.ElementIsVisible(By.CssSelector(".alert.alert-success")));
            Assert.IsTrue(saveIsSuccesful.Displayed);
            Console.WriteLine(saveIsSuccesful.Text);

            driver.Quit();
        }
    }
}

File: java/src/test/java/testcases/chapterSix/ShortCutAdjustmentTest.java
package testcases.chapterSix;

import org.assertj.core.api.Assertions;
import org.openqa.selenium.By;
import org.testng.annotations.Test;
import testcases.TestShopScenario;

public class ShortCutAdjustmentTest extends TestShopScenario{

    @Test
    public void shortCutAdjustment(){
        // Click on the login link
        driver.findElement(By.className("login")).click();

        // Enter username
        driver.findElement(By.id("email")).sendKeys("bootcamper@feelthepain.com");

        // Enter password
        driver.findElement(By.id("passwd")).sendKeys("1qazxsw2");

        // Click the Log in button
        driver.findElement(By.id("SubmitLogin")).click();

        //Go to the homepage
        driver.findElement(By.cssSelector("img.logo")).click();

        //personal info page can be found below
        driver.findElement(By.cssSelector("a[title='Manage my personal information']")).click();


        //Retrieve the current mail adres
        String currentFirstName = driver.findElement(By.id("firstname")).getAttribute("value");
        System.out.println(currentFirstName);
        String changedName;
        if (currentFirstName.equals("Sergeant")){
            driver.findElement(By.id("firstname")).clear();
            driver.findElement(By.id("firstname")).sendKeys("General");
            changedName = "General";
        }
        else if (currentFirstName.equals("General")){
            driver.findElement(By.id("firstname")).clear();
            driver.findElement(By.id("firstname")).sendKeys("Sergeant");
            changedName = "Sergeant";
        }
        else{
            driver.findElement(By.id("firstname")).clear();
            driver.findElement(By.id("firstname")).sendKeys("Seargent");
            changedName = "Seargent";
        }

        driver.findElement(By.id("old_passwd")).sendKeys("1qazxsw2");
        driver.findElement(By.cssSelector("button[name='submitIdentity']")).click();

        //Check if the empty message appears
        Assertions.assertThat(driver.findElement(By.cssSelector(".alert.alert-success")).getText())
                .as("Validate if succesfull message appears")
                .isEqualTo("Your personal information has been successfully updated.");

        driver.findElement(By.cssSelector(".account>span")).click();

        //Click the personal info icon
        driver.findElement(By.cssSelector(".icon-user")).click();

        //Check the new name
        //By running the test twice the console outputs should switch!
        System.out.println(driver.findElement(By.id("firstname")).getAttribute("value"));
        Assertions.assertThat(driver.findElement(By.id("firstname")).getAttribute("value"))
                .as("Validate name change")
                .isEqualTo(changedName);

    }
}
